### PR TITLE
feat: add strain doctype to new bloomtrace architecture with latest c…

### PR DESCRIPTION
**Pull request by harshmule-epi**
> TASK: Create strain doctype to the new bloomtrace architecture
[Task ID](https://bloomstack.com/desk#Form/Task/TASK-2021-00242)

### CODE REVIEW

File: erpnext/agriculture/doctype/strain/strain.py [hunk 1]
```diff
@@ -32,6 +37,52 @@ def validate_strain_tasks(self):
...
+		Args: 
+		
+		Returns: 
+			Successful Message to the User
+		"""
+		now = datetime.now()
+		try:
+			data = self.as_dict()
+			data['owner'] = "neil@bloomstack.com"
+			data['modified_by'] = "neil@bloomstack.com"
```
Comment (rvpasari) [inline erpnext/agriculture/doctype/strain/strain.py]: this is hard coded, won't work. Has to be the individual who created the file

Comment (rvpasari) [reply erpnext/agriculture/doctype/strain/strain.py]: the modified by details should be in "data"

File: erpnext/agriculture/doctype/strain/strain.py [hunk 2]
```diff
@@ -32,6 +37,52 @@ def validate_strain_tasks(self):
...
+				"license_number": "A12-0000015-LIC",
+				"Hello": "World",
+				"rest_method": "POST",
+				"environment": "sandbox",
+				"rest_operation": "Strains Create",
+				"doctype": "Strain",
+				"doctype_data": data
+			}
+			# create a strain on the bloomtrace
+			bloomtrace_response = requests.post('https://bl2qu9obqb.execute-api.ap-south-1.amazonaws.com/dev/doctype/createstrain', json=request_data)
```
Comment (rvpasari) [inline erpnext/agriculture/doctype/strain/strain.py:66]: the url has to come from bloomtrace settings doctype

**Review by rvpasari — CHANGES_REQUESTED** [review]

File: erpnext/agriculture/doctype/strain/strain.py (hunk 1 above)
Comment (harshmule-epi) [reply erpnext/agriculture/doctype/strain/strain.py]: it's there but it's coming of the logged in user, which is harsh mule/administrator, and this user is used to get the user_key from admin portal (as per the code) and non of this user can get me the user_key, hence metrc is failing, only neil@bloomstack.com is working.

File: erpnext/agriculture/doctype/strain/strain.py (hunk 2 above)
Comment (harshmule-epi) [reply erpnext/agriculture/doctype/strain/strain.py:66]: yes, creating a PR for that

File: erpnext/agriculture/doctype/strain/strain.py (hunk 1 above)
Comment (rvpasari) [reply erpnext/agriculture/doctype/strain/strain.py]: create a user with your email on the admin portal, add the same keys as neil's

Comment (harshmule-epi) [reply erpnext/agriculture/doctype/strain/strain.py]: Cool

File: erpnext/agriculture/doctype/strain/strain.py [hunk 3]
```diff
@@ -32,6 +37,51 @@ def validate_strain_tasks(self):
...
+	
+	def update_to_bloomtrace(self):
+		"""
+		Create a new strain on Bloomtrace via Bloomstack.
+		Args: 
+		
+		Returns: 
+			Successful Message to the User
+		"""
+		now = datetime.now()
```
Comment (vishdha) [inline erpnext/agriculture/doctype/strain/strain.py:49]:
```suggestion

```
Unused declaration

**Review by vishdha — CHANGES_REQUESTED** [review]

**Pull request closed without merging**